framework/doc/content/documentation/systems/MeshModifiers/BreakMeshByBlock.md

```diff
@@ -0,0 +1,36 @@
+# Break Mesh By Block
+
+!syntax description /MeshModifiers/BreakMeshByBlock
+
+This class implement a methods to split a monolithic mesh by blocks and follows the
+method proposed by VP Nguyen [cite:Nguyen2014]
+
+To split the mesh nodes belonging to a boundary between 2 blocks are duplicated, the mesh is split and a new sideset is added. The new sideset is the boundary where the Interface lives.
+
+
+
+As an option, the interface can be split into $N$ different sidesets. $N$ is the number of adjacent block pairs. This is achieved by setting  `split_interface=true`. This is useful when modeling interfaces with different parameters.
+
+## Example Input File Syntax
+
+
+
+### Single interface
+
+!listing test/tests/mesh_modifiers/break_mesh_by_blocks/2D/TestBreakMesh_2DJunction_Auto.i block=MeshModifiers
+
+### Multiple interfaces
+
+When `split_interface=true` the new generated interface is splitted by block pairs:
+
+!listing test/tests/mesh_modifiers/break_mesh_by_blocks/2D/TestBreakMesh_2DJunction_splitTrue_Auto.i block=MeshModifiers
+
+The naming convention for the interface is `bMi_bSj`, where $i$ and $j$ represents the original blocks ids.
+This convention also assumes that the master interface $i$ is set on the block with lower id, while the slave interface $j$ is always on the block with higher id.
```

Comment:
Is this (the block numbering dependency) a requirement of this object to work?

Reply:
Derek, if you are talking about `bMi_bSj` it's just the way the object names the new generate sidesets for split_interface=true. This is an output.

I made the choice to generate sidesets always on the MasterSide (lower block id) because you don't need complicated checks and each block can have as many neighbours as you want.

Right now you can't selectively choose blocks to split, the object will just make interface s between all the neighboring block (selective splitting is not trivial)


Does this answer your question?

Reply:
@friedmud I was thinking to your question while driving home, and I think this is the answer. No we don't need this convention`bMi_bSj`. If block names are guaranteed to be unique we can use block names. For instance if we have two blocks `steel` and `concrete` with ids `1` and `2`,respectively, we can name the interface something like `steel_concrete_interface`.
Is this what you want? Also do you like the suffix `_interface` or you want something more descriptive like `_splitinterface`.